mixnet/src/config.rs

```diff
@@ -0,0 +1,22 @@
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Config {
+    pub addr: SocketAddr,
+    // TODO: add pubkeys
+    // TODO: find better ways to handle topology
+    pub topology: Vec<SocketAddr>,
```

Comment:
The important point here would be decoupling the `libp2p::PeerId` with `nomos::NodeId`, so that anyone cannot derive `nomos::NodeId` that is probably associated with stakes.

Reply:
I just talked with @danielSanchezQ about trying this DHT idea. I'll put a super simple prototype on top of this PR soon.

Reply:
Just implemented a naive approach of managing the topology: [84ec246](https://github.com/logos-co/nomos-node/pull/288/commits/84ec246331c78f4d6247551c3360e428b05bc66f). This doesn't actually use DHT. Instead, each node broadcasts their external IP address and their public key via gossipsub, so that receiving nodes can update the mixnet topology that they hold in their memory.